conda_build/conda_interface.py

```diff
@@ -165,7 +165,8 @@ def default(self, obj):
     cc_conda_build = context.conda_build if hasattr(context, 'conda_build') else {}
 
     # disallow softlinks.  This avoids a lot of dumb issues, at the potential cost of disk space.
-    os.environ['CONDA_ALLOW_SOFTLINKS'] = 'false'
+    # HINT: Keep the str() for py2 running on windows, see https://github.com/conda/conda-build/issues/1865
+    os.environ[str('CONDA_ALLOW_SOFTLINKS')] = str('false')
```

Comment:
Yep, valid. But I think

- py2 compat removal is IMHO at least three years away from now. Who knows whether this lines are still alive then. To me it looks like programming for the stock.
- One won't fetch _all_ the PY2 issues while grepping for PY2, but it may be a very good idea, to grep / check the AST for things like "str(" anyway.
- and now you have two lines doing mainly the same, so you have to keep an eye, that always both lines get modified.

- Just my thoughts -